src/Transports/MassTransit.KafkaIntegration/Contexts/ConsumerLockContext.cs

```diff
@@ -134,8 +139,33 @@ void CommitIfRequired()
                     return;
 
                 var offset = _offset + 1;
+                var partitionOffset = new TopicPartitionOffset(_partition, offset);
+
+                if (_observers.Count > 0)
+                    TaskUtil.Await(() => _observers.PreCheckpoint(new KafkaCheckpointEvent(partitionOffset)));
+
                 LogContext.Debug?.Log("Partition: {PartitionId} updating checkpoint with offset: {Offset}", _partition, offset);
-                _consumer.Commit(new[] {new TopicPartitionOffset(_partition, offset)});
+                _consumer.Commit(new[] {partitionOffset});
+
+                if (_observers.Count > 0)
+                    TaskUtil.Await(() => _observers.PostCheckpoint(new KafkaCheckpointEvent(partitionOffset)));
```

Comment:
If there isn't an async context here, don't make the observers async? Using TaskUtil.Await is always a bad idea except when it absolutely can't be avoided. It's expensive. Very expensive.

Reply:
I moved `partition.Close` to `TaskUtil.Await` it won't be called so often. Which make sense